src/main/java/org/dsa/iot/calendar/Actions.java

```diff
@@ -388,44 +383,39 @@ public void handle(ActionResult actionResult) {
-                            }
-                            Date startDate = new SimpleDateFormat(BaseCalendar.DATE_PATTERN).parse(dates[0]);
-                            Date endDate = new SimpleDateFormat(BaseCalendar.DATE_PATTERN).parse(dates[1]);
-                            event.setStart(startDate);
-                            event.setEnd(endDate);
-                            event.setLocation(location);
-                            Node editEventNode = actionResult.getNode().getParent();
-                            editEventNode.setDisplayName(title);
-                            editEventNode.getChild("description").setValue(new Value(desc));
-                            editEventNode.getChild("start").setValue(new Value(dates[0]));
-                            editEventNode.getChild("end").setValue(new Value(dates[1]));
-                            editEventNode.getChild("location").setValue(new Value(location));
-                            if (calendar.supportsMultipleCalendars()) {
-                                String calendarId = editEventNode.getChild("calendarId").getValue().getString();
-                                event.setCalendar(new DSAIdentifier(calendarId,
-                                        editEventNode.getChild("calendar").getValue().getString()));
-                            }
-                            calendar.deleteEvent(editEventNode.getName(), false);
-                            calendar.createEvent(event);
-                            actionResult.getTable().addRow(Row.make(new Value(true)));
-                        } catch (Exception e) {
-                            e.printStackTrace();
-                            actionResult.getTable().addRow(Row.make(new Value(false)));
+                        event.setLocation(location);
+                        Node editEventNode = actionResult.getNode().getParent();
+                        editEventNode.setDisplayName(title);
+                        editEventNode.getChild("description").setValue(new Value(desc));
+                        editEventNode.getChild("start").setValue(new Value(dates[0]));
+                        editEventNode.getChild("end").setValue(new Value(dates[1]));
+                        editEventNode.getChild("location").setValue(new Value(location));
+                        if (calendar.supportsMultipleCalendars()) {
+                            String calendarId = editEventNode.getChild("calendarId").getValue().getString();
+                            event.setCalendar(new DSAIdentifier(calendarId,
+                                    editEventNode.getChild("calendar").getValue().getString()));
                         }
+                        calendar.deleteEvent(editEventNode.getName(), false);
+                        calendar.createEvent(event);
+                        actionResult.getTable().addRow(Row.make(new Value(true)));
+                    } catch (Exception e) {
+                        e.printStackTrace();
```

Comment:
Probably not good to print the stacktrace to stdio, need to use a logger or return error gracefully to the action.